hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/compactions/StripeCompactionPolicy.java

```diff
@@ -125,6 +125,21 @@ public StripeCompactionRequest selectCompaction(StripeInformationProvider si,
       return request;
     }
 
+    if (forceMajor && isUserCompaction) {
+      if (allFiles == null || allFiles.isEmpty()) {
+        LOG.debug("There is no store file to force a major compaction.");
+        return null;
+      }
+      LOG.debug("The user enforces a major compaction; compacting all files");
+      long targetKvs = estimateTargetKvs(allFiles, config.getInitialCount()).getFirst();
+      SplitStripeCompactionRequest request =
```

Comment:
This will compact all files, not only the files in a single stripe?

Reply:
Yes, if it is a user-triggered forced marjor compact, it will perform the compact of all files in the entire region, and for periodic major compact, it will select a certain Stripe to compact

Reply:
Will this break the stripe assumption, for example, size based stripe? Anyway, I'm not very familiar with stripe compaction...

Reply:
> Will this break the stripe assumption, for example, size based stripe? Anyway, I'm not very familiar with stripe compaction...

Probably not, the logic of the forced major compact of the whole region reuses the compact logic of the reference Hfile, I tested it and found that after major compact, one file is generated in each Stripe, and the number of files is basically the same as the initialized stripe count, and the file size under each Stripe is also relatively average